### PR TITLE
List database session recordings on WebUI and tsh

### PR DESCRIPTION
**Pull request by gabrielcorado**
Part of [RFD 0171](https://github.com/gravitational/teleport/blob/master/rfd/0171-database-session-playback.md) implementation.

It enables the recordings list to show database session recordings. Also, add a "Play" button for supported database protocols.

The required changes were:
- Add `StartTime` and `EndTime` to the `DatabaseSessionEnd` audit event. This enables the WebUI to calculate the recording duration correctly, present it, and configure the player progress bar.
- Add a `makeDatabaseRecording` function, which takes the session end event and generates the proper metadata (similar to other recordings).
- Add a new configuration for the WebUI only to present a play button for supported database protocols.

<details>
<summary>WebUI listing screenshot</summary>

Screenshot 2024-07-05 at 19 23 20
</details>

<details>
<summary>WebUI play recording video</summary>

https://github.com/gravitational/teleport/assets/2467749/0d20159d-f718-49a7-8fd2-dc70fce134c5
</details>

<details>
<summary>`tsh` recordings</summary>

```bash
❯ tsh recordings ls
ID                                   Type     Participants Hostname Timestamp
------------------------------------ -------- ------------ -------- ------------------------
3861d055-6c34-4129-a604-c0d4ccf2d691 mysql    alice        mysql    Jul  5 2024 22:21:40 UTC
2b303bf4-56d5-46a3-bb44-833975bc1eb1 postgres alice        postgres Jul  5 2024 19:09:42 UTC

```
</details>

changelog: Add database session recordings to the session recording listing (WebUI and `tsh recordings ls`).

### CODE REVIEW

File: tool/common/common.go [hunk 1]
```diff
@@ -72,6 +72,12 @@ func (e *SessionsCollection) WriteText(w io.Writer) error {
...
 			hostname = session.DesktopName
 			timestamp = session.GetTime().Format(constants.HumanDateFormatSeconds)
+		case *events.DatabaseSessionEnd:
+			id = session.GetSessionID()
+			typ = session.DatabaseProtocol
+			participants = session.GetUser()
+			hostname = session.DatabaseName
+			timestamp = session.GetTime().Format(constants.HumanDateFormatSeconds)
 		default:
 			log.Warn(trace.BadParameter("unsupported event type: expected SessionEnd or WindowsDesktopSessionEnd: got: %T", event))
```
Comment (zmb3) [inline tool/common/common.go:82]: Update this error message.

**Review by zmb3 — APPROVED** [review]

File: web/packages/teleport/src/services/recordings/makeRecording.ts [hunk 2]
```diff
@@ -119,4 +123,42 @@ function formatDuration(startDateString: string, stopDateString: string) {
...
+  let { duration, durationText } = formatDuration(
+    session_start,
+    session_stop
+  );
+
+  // Older database session recordings won't have start/stop fields. For those
+  // recordings we set the duration to the smallest number so we can still
+  // them.
+  // As a side effect, the progress bar to not work properly, showing always as
+  // completed. Also, navigating through it won't work.
```
Comment (greedy52) [inline web/packages/teleport/src/services/recordings/makeRecording.ts:145]: nit:
```suggestion
  // Older database session recordings won't have start/stop fields. For those
  // recordings we set the duration to the smallest number so we can still
  // play them.
  // As a side effect, the progress bar does not work properly, showing always as
  // completed. Also, navigating through it won't work.
```

File: lib/web/apiserver.go [hunk 3]
```diff
@@ -1682,7 +1683,8 @@ func (h *Handler) getWebConfig(w http.ResponseWriter, r *http.Request, p httprou
 		MobileDeviceManagement: clusterFeatures.GetMobileDeviceManagement(),
 		JoinActiveSessions:     clusterFeatures.GetJoinActiveSessions(),
 		// TODO(mcbattirola): remove isTeam when it is no longer used
-		IsTeam: isTeam,
+		IsTeam:                    isTeam,
+		PlayableDatabaseProtocols: player.SupportedDatabaseProtocols,
```
Comment (greedy52) [inline lib/web/apiserver.go:1687]: nice 👍

**Review by greedy52 — APPROVED** [review]

File: web/packages/teleport/src/services/recordings/makeRecording.ts [hunk 4]
```diff
@@ -119,4 +123,42 @@ function formatDuration(startDateString: string, stopDateString: string) {
...
+    durationText = '-';
+  }
+
+  return {
+    duration,
+    durationText,
+    sid,
+    createdDate: time,
+    users: user,
+    hostname: db_name,
```
Comment (greedy52) [inline web/packages/teleport/src/services/recordings/makeRecording.ts:157]: Should this be the database service name?
```suggestion
    hostname: db_service_name,
```

 Ideally, we should db_user, db_service, and db_name info some where in this table. Or do we print out db_user and db_name info when playing?

Comment (gabrielcorado) [reply web/packages/teleport/src/services/recordings/makeRecording.ts:157]: > Ideally, we should show db_user, db_service, and db_name info some where in this table.

That's correct, I'll update it.

> Or do we print out db_user and db_name info when playing?

I'm adding this information to the session recording itself. (I'll update the other PR soon).

**Pull request merged**